src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -1184,7 +1204,13 @@ private void GenerateInheritanceDiscriminator(Type type, JsonSchema schema, Json
         private object TryGetInheritanceDiscriminatorConverter(Type type)
         {
             var typeAttributes = type.GetTypeInfo().GetCustomAttributes(false).OfType<Attribute>();
-
+#if NET7_0_OR_GREATER
+            dynamic jsonPolymorphicAttribute = typeAttributes.FirstAssignableToTypeNameOrDefault(nameof(System.Text.Json.Serialization.JsonPolymorphicAttribute), TypeNameStyle.Name);
+            if (jsonPolymorphicAttribute != null)
+            {
+                return jsonPolymorphicAttribute;
```

Comment:
Not sure if it's a bit "too dirty" to use the JsonPolymorphicAttribute as inheritance converter.